tests/components/homeassistant_sky_connect/test_init.py

```diff
@@ -58,3 +88,129 @@ async def test_config_entry_migration_v2(hass: HomeAssistant) -> None:
     }
 
     await hass.config_entries.async_unload(config_entry.entry_id)
+
+
+async def test_setup_fails_on_missing_usb_port(hass: HomeAssistant) -> None:
+    """Test setup failing when the USB port is missing."""
+
+    config_entry = MockConfigEntry(
+        domain=DOMAIN,
+        unique_id="some_unique_id",
+        data={
+            "description": "SkyConnect v1.0",
+            "device": "/dev/serial/by-id/usb-Nabu_Casa_SkyConnect_v1.0_9e2adbd75b8beb119fe564a0f320645d-if00-port0",
+            "vid": "10C4",
+            "pid": "EA60",
+            "serial_number": "3c0ed67c628beb11b1cd64a0f320645d",
+            "manufacturer": "Nabu Casa",
+            "product": "SkyConnect v1.0",
+            "firmware": "ezsp",
+            "firmware_version": "7.4.4.0",
+        },
+        version=1,
+        minor_version=3,
+    )
+
+    config_entry.add_to_hass(hass)
+
+    # Set up the config entry
+    with patch(
+        "homeassistant.components.homeassistant_sky_connect.os.path.exists"
+    ) as mock_exists:
+        mock_exists.return_value = False
+        await hass.config_entries.async_setup(config_entry.entry_id)
+        await hass.async_block_till_done()
+
+        # Failed to set up, the device is missing
+        assert config_entry.state == ConfigEntryState.SETUP_RETRY
+
+        mock_exists.return_value = True
+        async_fire_time_changed(hass, dt_util.now() + timedelta(seconds=30))
+        await hass.async_block_till_done(wait_background_tasks=True)
+
+        # Now it's ready
+        assert config_entry.state == ConfigEntryState.LOADED
+
+
+async def test_usb_device_reactivity(
+    hass: HomeAssistant, hass_ws_client: WebSocketGenerator
+) -> None:
+    """Test setting up USB monitoring."""
+
+    # Disable the inotify subsystem to allow us to trigger rescans manually
+    with patch(
+        "homeassistant.components.usb.AIOUSBWatcher",
+        side_effect=InotifyNotAvailableError,
+    ):
```

Comment:
This patcher should be provided by the usb integration so when the implementation details change in usb this test won't break

Reply:
I'll make a PR to break this out then.